Saving record with non-Latin characters in title throws Internal Server Error

Fixed in #4131
Related to #3704

<!-- What problem are we solving? What does the experience look like today? What are the symptoms? -->
Saving record with non-Latin characters in title throws Internal Server Error as of 11/13/2020.
### Evidence / Screenshot (if possible)
<img width="765" alt="Screenshot 2020-11-13 at 18 34 07" src="https://user-images.githubusercontent.com/17739465/99102783-dd422880-25de-11eb-84b3-f95c44a9b32c.png">

### Relevant url?
<!-- `https://openlibrary.org/books/OL...M/edit

### Steps to Reproduce
<!-- What steps caused you to find the bug? -->
1. Go to any edition
2. Edit title by adding non-latin characters
3. Save

<!-- What actually happened after these steps? What did you expect to happen? -->
* Actual: Internal server error, edit not saved
* Expected: Save edits.

### Details

- **Logged in (Y/N)?** Y
- **Browser type/version?** Chrome Version 86.0.4240.193 (Official Build) (x86_64)
- **Operating system?** Mac OS Big Sur
- **Environment (prod/dev/local)?** prod
<!-- If not sure, put prod -->

### Proposal & Constraints
<!-- What is the proposed solution / implementation? Is there a precedent of this approach succeeding elsewhere? -->

### Related files
<!-- Files related to this issue; this is super useful for new contributors who might want to help! If you're not sure, leave this blank; a maintainer will add them. -->

### Stakeholders
<!-- @ tag stakeholders of this bug -->
@seabelis

```
[2020-11-15 10:36:11 +0000] [216] [ERROR] Error handling request /books/OL21695881M/Dictionnaire_des_ide%CC%81es_rec%CC%A7ues/edit
Traceback (most recent call last):
  File "/home/openlibrary/.pyenv/versions/3.8.6/lib/python3.8/site-packages/gunicorn/workers/sync.py", line 134, in handle
    self.handle_request(listener, req, client, addr)
  File "/home/openlibrary/.pyenv/versions/3.8.6/lib/python3.8/site-packages/gunicorn/workers/sync.py", line 181, in handle_request
    resp.write(item)
  File "/home/openlibrary/.pyenv/versions/3.8.6/lib/python3.8/site-packages/gunicorn/http/wsgi.py", line 326, in write
    self.send_headers()
  File "/home/openlibrary/.pyenv/versions/3.8.6/lib/python3.8/site-packages/gunicorn/http/wsgi.py", line 322, in send_headers
    util.write(self.sock, util.to_bytestring(header_str, "latin-1"))
  File "/home/openlibrary/.pyenv/versions/3.8.6/lib/python3.8/site-packages/gunicorn/util.py", line 564, in to_bytestring
    return value.encode(encoding)
UnicodeEncodeError: 'latin-1' codec can't encode character '\u0301' in position 249: ordinal not in range(256)
```

--- a/openlibrary/plugins/upstream/addbook.py
+++ b/openlibrary/plugins/upstream/addbook.py
@@ -782,7 +782,7 @@ class book_edit(delegate.page):
             else:
                 add_flash_message("info", utils.get_message("flash_book_updated"))
 
-            raise web.seeother(edition.url())
+            raise web.seeother(urllib.parse.quote(edition.url()))
         except ClientException as e:
             add_flash_message('error', e.args[-1] or e.json)
             return self.GET(key)

